cli/cmd/top.go

```diff
@@ -62,11 +79,145 @@ type tableRow struct {
 	failures    int
 }
 
-const headerHeight = 3
+func (r tableRow) merge(other tableRow) tableRow {
+	r.count += other.count
+	if other.best.Nanoseconds() < r.best.Nanoseconds() {
+		r.best = other.best
+	}
+	if other.worst.Nanoseconds() > r.worst.Nanoseconds() {
+		r.worst = other.worst
+	}
+	r.last = other.last
+	r.successes += other.successes
+	r.failures += other.failures
+	return r
+}
+
+const (
+	headerHeight  = 3
+	columnSpacing = 2
+)
 
 var (
-	columnNames  = []string{"Source", "Destination", "Method", "Path", "Count", "Best", "Worst", "Last", "Success Rate"}
-	columnWidths = []int{23, 23, 10, 37, 6, 6, 6, 6, 3}
+	sourceColumn = tableColumn{
+		header:   "Source",
+		width:    23,
+		key:      true,
+		display:  true,
+		flexible: true,
+		value: func(r tableRow) string {
+			return r.source
+		},
+	}
+
+	destinationColumn = tableColumn{
+		header:   "Destination",
+		width:    23,
+		key:      true,
+		display:  true,
+		flexible: true,
+		value: func(r tableRow) string {
+			return r.destination
+		},
+	}
+
+	methodColumn = tableColumn{
+		header:   "Method",
+		width:    10,
+		key:      true,
+		display:  true,
+		flexible: false,
+		value: func(r tableRow) string {
+			return r.method
+		},
+	}
+
+	pathColumn = tableColumn{
+		header:   "Path",
+		width:    37,
+		key:      true,
+		display:  true,
+		flexible: true,
+		value: func(r tableRow) string {
+			return r.path
+		},
+	}
+
+	routeColumn = tableColumn{
+		header:   "Route",
+		width:    47,
+		key:      false,
+		display:  false,
+		flexible: true,
+		value: func(r tableRow) string {
+			return r.route
+		},
+	}
+
+	countColumn = tableColumn{
+		header:     "Count",
+		width:      6,
+		key:        false,
+		display:    true,
+		flexible:   false,
+		rightAlign: true,
+		value: func(r tableRow) string {
+			return strconv.Itoa(r.count)
+		},
+	}
+
+	bestColumn = tableColumn{
+		header:     "Best",
+		width:      6,
+		key:        false,
+		display:    true,
+		flexible:   false,
+		rightAlign: true,
+		value: func(r tableRow) string {
+			return formatDuration(r.best)
+		},
+	}
+
+	worstColumn = tableColumn{
+		header:     "Worst",
+		width:      6,
+		key:        false,
+		display:    true,
+		flexible:   false,
+		rightAlign: true,
+		value: func(r tableRow) string {
+			return formatDuration(r.worst)
+		},
+	}
+
+	lastColumn = tableColumn{
+		header:     "Last",
+		width:      6,
+		key:        false,
+		display:    true,
+		flexible:   false,
+		rightAlign: true,
+		value: func(r tableRow) string {
+			return formatDuration(r.last)
+		},
+	}
+
+	successRateColumn = tableColumn{
+		header:     "Success Rate",
+		width:      12,
+		key:        false,
+		display:    true,
+		flexible:   false,
+		rightAlign: true,
+		value: func(r tableRow) string {
+			return fmt.Sprintf("%.2f%%", 100.0*float32(r.successes)/float32(r.successes+r.failures))
+		},
+	}
+
+	columns = []*tableColumn{
```

Comment:
i realize we already had this pattern, but this introduces a lot of global vars into the `cmd` package. consider hiding it in a function, something like...

```go
func mkColumns(options topOptions) []tableColumn {
	sourceColumn := tableColumn{...}
	...
	return []tableColumn{sourceColumn, ...}
}

...

return getTrafficByResourceFromAPI(os.Stdout, validatedPublicAPIClient(time.Time{}), req, mkColumns(options))
```

... you could even go a step further and create a new `topTable` type, something like...

```go
type topTable struct{
	columns []tableColumn
	rows []tableRow
}

func newTopTable(options topOptions) (*topTable) {}
func (t *topTable) insert(row tableRow) {}
func (t *topTable) render() {}
```